src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts

```diff
@@ -215,13 +217,18 @@ export class PivotSidePanelStore extends SpreadsheetStore {
     const { columns, rows } = definition;
     const columnsWithGranularity = deepCopy(columns);
     const rowsWithGranularity = deepCopy(rows);
-    const unusedGranularities = this.getUnusedDateTimeGranularities(fields, definition);
+    const unusedGranularities = this.getUnusedDateTimeGranularities(fields, definition, undefined);
     for (const dimension of columnsWithGranularity.concat(rowsWithGranularity)) {
       const fieldType = fields[dimension.name]?.type;
       if ((fieldType === "date" || fieldType === "datetime") && !dimension.granularity) {
-        const granularity = unusedGranularities[dimension.name]?.values().next().value || "year";
```

Comment:
after thinking about it a second time and discussing with @pro-odoo . I'd rather live with the simpler code even if we have the strange behavior where "month" becomes "year". It's ok IMO, the user is updating and breaking the dimensions anyway.
Not that the code is really difficult, but the flow for which it matters is tordu (the "why" is this code like that).

Reply:
Sure. I added a comment in the test to say it's foireux